code/modules/power/solar.dm

```diff
@@ -13,38 +16,46 @@
 	max_integrity = 150
 	integrity_failure = 0.33
 
-	var/id = 0
-	max_integrity = 150
-	integrity_failure = 50
-	var/obscured = 0
-	var/sunfrac = 0
-	var/adir = SOUTH // actual dir
-	var/ndir = SOUTH // target dir
-	var/turn_angle = 0
-	var/obj/machinery/power/solar_control/control = null
+	var/id
+	var/obscured = FALSE
+	var/sunfrac = 0 //[0-1] measure of obscuration -- multipllier against power generation
+	var/azimuth_current = 0 //[0-360) degrees, which direction are we facing?
+	var/azimuth_target = 0 //same but what way we're going to face next time we turn
+	var/obj/machinery/power/solar_control/control
+	var/needs_to_turn = TRUE //do we need to turn next tick?
+	var/needs_to_update_solar_exposure = TRUE //do we need to call update_solar_exposure() next tick?
+	var/obj/effect/overlay/panel
 
 /obj/machinery/power/solar/Initialize(mapload, obj/item/solar_assembly/S)
 	. = ..()
+	panel = new()
+#if DM_VERSION >= 513
+	panel.vis_flags = VIS_INHERIT_ID | VIS_INHERIT_ICON | VIS_INHERIT_PLANE
+	vis_contents += panel
```

Comment:
Lets not use vis contents if we can.
A better solution would be to use an overlay with a rotation applied in update_icon() and then calling update_icon() to update that overlay when it moves.
Or alternatively, just having the solar panel as an object on top of this object.